compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp

```diff
@@ -82,6 +82,12 @@
 #    include <sys/personality.h>
 #  endif
 
+#  if SANITIZER_ANDROID && __ANDROID_API__ < 35
+// The weak `strerrorname_np` (introduced in API level 35) definition,
+// allows for checking the API level at runtime.
```

Comment:
What's wrong with https://cs.android.com/android/platform/superproject/main/+/main:bionic/libc/include/android/api-level.h;l=214;drc=02ce401d1e2b31586c94c8b6999364bbbce27388?

Reply:
Does not work in `.preinit_array`.
```fish
❯ echo '#include <android/api-level.h>'\n'static int api_level;static void preinit_fn() {api_level=android_get_device_api_level();} void (*preinit[])()__attribute((section(".preinit_array")))={&preinit_fn};int main(){__builtin_printf("api_level on preinit: %i. on main: %i", api_level, android_get_device_api_level());}' | cc -x c -;./a.out
api_level on preinit: -1. on main: 27⏎         
```

Reply:
Yeah, we can't use anything in libc because we are running before libc is relocated. we can add this in a comment for clarity.